src/libmongoc/src/mongoc/mongoc-crypt.c

```diff
@@ -41,6 +44,12 @@ struct __mongoc_crypt_t {
    /// credentials.
    bson_t kms_providers;
    mc_kms_credentials_callback creds_cb;
+
+   /// The most recently auto-acquired Azure token, on null if it was destroyed
+   /// or not yet acquired.
+   mcd_azure_access_token azure_token;
+   /// The time point at which the `azure_token` was acquired.
+   mcd_time_point azure_token_issued_at;
```

Comment:
It could, but you can't determine when a token was issued based only on the token json data (except for the issue time according to the server, which is a Unix timestamp, and would therefore be subject to local clock offsets.)